source/renderer/app/components/staking/delegation-setup-wizard/DelegationStepsChooseStakePoolDialog.js

```diff
@@ -0,0 +1,282 @@
+import Dialog from '../../widgets/Dialog';
+import { StakePoolsList } from '../stake-pools/StakePoolsList';
+import { StakePoolsSearch } from '../stake-pools/StakePoolsSearch';
+import styles from './DelegationStepsChooseStakePoolDialog.scss';
+import checkmarkImage from '../../../assets/images/check-w.inline.svg';
+import type { StakePool } from '../../../api/staking/types';
+
+const messages = defineMessages({
+  title: {
+    id: 'staking.delegationSetup.chooseStakePool.step.dialog.title',
+    defaultMessage: '!!!Delegation Setup',
+    description:
+      'Title "Delegation Setup" on the delegation setup "choose stake pool" dialog.',
+  },
+  description: {
+    id: 'staking.delegationSetup.chooseStakePool.step.dialog.description',
+    defaultMessage:
+      '!!!Choose a stake pool to which you would like to delegate.',
+    description:
+      'Description on the delegation setup "choose stake pool" dialog.',
+  },
+  delegatedPoolsLabel: {
+    id:
+      'staking.delegationSetup.chooseStakePool.step.dialog.delegatedPoolsLabel',
+    defaultMessage: '!!!Stake pools you are already delegating to:',
+    description:
+      '"Delegated Pools" section label on the delegation setup "choose stake pool" dialog.',
+  },
+  searchInputLabel: {
+    id: 'staking.delegationSetup.chooseStakePool.step.dialog.searchInput.label',
+    defaultMessage: '!!!Or search for a stake pool:',
+    description:
+      'Search "Pools" input label on the delegation setup "choose stake pool" dialog.',
+  },
+  searchInputPlaceholder: {
+    id:
+      'staking.delegationSetup.chooseStakePool.step.dialog.searchInput.placeholder',
+    defaultMessage: '!!!Search stake pools',
+    description:
+      'Search "Pools" input placeholder on the delegation setup "choose stake pool" dialog.',
+  },
+  continueButtonLabel: {
+    id:
+      'staking.delegationSetup.chooseStakePool.step.dialog.continueButtonLabel',
+    defaultMessage: '!!!Continue',
+    description:
+      'Label for continue button on the delegation setup "choose stake pool" dialog.',
+  },
+  stepIndicatorLabel: {
+    id:
+      'staking.delegationSetup.chooseStakePool.step.dialog.stepIndicatorLabel',
+    defaultMessage: '!!!STEP {currentStep} OF {totalSteps}',
+    description:
+      'Step indicator labe on the delegation setup "choose wallet" step dialog.',
+  },
+  selectPoolPlaceholder: {
+    id:
+      'staking.delegationSetup.chooseStakePool.step.dialog.selectPoolPlaceholder',
+    defaultMessage: '!!!POOL',
+    description:
+      'Selected pool box placeholder on the delegation setup "choose wallet" step dialog.',
+  },
+});
+
+type Props = {
+  stepsList: Array<string>,
+  stakePoolsDelegatingList: Array<StakePool>,
+  stakePoolsList: Array<StakePool>,
+  onOpenExternalLink: Function,
+  currentTheme: string,
+  onClose: Function,
+  onContinue: Function,
+  onBack: Function,
+};
+
+type State = {
+  searchValue: string,
+  selectedList?: ?string,
+  flipHorizontal: boolean,
```

Comment:
The `flipHorizontal` and `flipVertical` state vars are not needed anymore. I'll remove them.

Reply:
@daniloprates nice work and thanks 🥇